### PR TITLE
Fix query failure with mixed distinct aggregation

### PR DESCRIPTION
**Pull request by wenleix**
Fix the issue that MixedDistinctAggregations optimizer fails when
GroupBy columns and aggregation columns have overlap.

### CODE REVIEW

**Comment by wenleix** [comment]
This addresses issue https://github.com/prestodb/presto/issues/7214

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 1]
```diff
@@ -210,6 +210,7 @@ public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Optional
             allSymbols.addAll(groupBySymbols);
             allSymbols.addAll(nonDistinctAggregateSymbols);
             allSymbols.add(distinctSymbol);
+            allSymbols = allSymbols.stream().distinct().collect(Collectors.toList());
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: We may want to use a Set for `allSymbols` and `groupByKeys`, instead.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 2]
```diff
@@ -227,26 +228,28 @@ public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Optional
             groupByKeys.addAll(groupBySymbols);
             groupByKeys.add(distinctSymbol);
             groupByKeys.add(groupSymbol);
+            groupByKeys = groupByKeys.stream().distinct().collect(Collectors.toList());
 
-            ImmutableMap.Builder aggregationOuputSymbolsMapBuilder = ImmutableMap.builder();
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: Move this typo fix to a separate commit.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 3]
```diff
@@ -344,12 +350,14 @@ private GroupIdNode createGroupIdNode(
             List<Symbol> group0 = new ArrayList<>();
             group0.addAll(groupBySymbols);
             group0.addAll(nonDistinctAggregateSymbols);
+            group0 = group0.stream().distinct().collect(Collectors.toList());
             groups.add(group0);
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: Same here. We may want to use a Set.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java (hunk 1 above)
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: @martint Do you mean use `Set<Symbol>` instead of `List<Symbol>` for `allSymbols` and `groupByKeys` ? :)

Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: @martint  Interestingly , after rebase, only deduplicating `groupByKeys` is required  -- if groupBy Column = Distinct Column, the distinct column will be automatically renamed to `col_2` . I assume some upstream changes in Planner did this rename.

I still converting `groupByKeys` into a `Set` since I think we shouldn't rely on the renaming behavior  in previous optimizer / planner.

**Comment by wenleix** [comment]
Revised :)

**Comment by wenleix** [comment]
Ping :)
